inventory/inventory.go

```diff
@@ -0,0 +1,75 @@
+package inventory
+
+import (
+	"github.com/opencontrol/compliance-masonry/models"
+	"github.com/opencontrol/compliance-masonry/tools/certifications"
+)
+
+// Inventory maintains the inventory of all the controls within a given workspace.
+type Inventory struct {
+	*models.OpenControl
+	masterControlList      []standardAndControl
+	actualSatifiedControls []models.Satisfies
+	MissingControlList     []standardAndControl
+}
+
+type standardAndControl struct {
+	control  string
+	standard string
+}
+
+func (c standardAndControl) String() string {
+	return c.standard + "@" + c.control
+}
+
+func (c standardAndControl) EqualToSatisfiedControl(other models.Satisfies) bool {
+	return c.standard == other.StandardKey && c.control == other.ControlKey
+}
+
+// Config contains the settings for how to compute the gap analysis
+type Config struct {
+	Certification  string
+	OpencontrolDir string
+}
+
+// ComputeGapAnalysis will compute the gap analysis and return the inventory of the controls for the
+// opencontrol workspace if successful. Otherwise, it will return a list of error messages.
+// TODO: fix the error return to return of type error. This was used because existing code returned that type
+// TODO: e.g. GetCertification
+func ComputeGapAnalysis(config Config) (Inventory, []string) {
+	certificationPath, messages := certifications.GetCertification(config.OpencontrolDir, config.Certification)
+	if certificationPath == "" {
+		return Inventory{}, messages
+	}
+	i := Inventory{OpenControl: models.LoadData(config.OpencontrolDir, certificationPath)}
+	if i.Certification == nil || i.Components == nil {
+		return Inventory{}, []string{"Unable to load data in " + config.OpencontrolDir + " for certification " + config.Certification}
+	}
+	// Master Controls
+	for certification, standard := range i.Certification.Standards {
+		for control, _ := range standard.Controls {
+			i.masterControlList = append(i.masterControlList, standardAndControl{standard: certification, control: control})
+		}
+	}
+	// Actual Controls
+	for _, components := range i.Components.GetAll() {
+		for _, satisfiedComponent := range *components.Satisfies {
+			i.actualSatifiedControls = append(i.actualSatifiedControls, satisfiedComponent)
+		}
+	}
+
+	// Missing controls
+	for _, masterControl := range i.masterControlList {
+		found := false
+		for _, actualControl := range i.actualSatifiedControls {
+			if masterControl.EqualToSatisfiedControl(actualControl) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			i.MissingControlList = append(i.MissingControlList, masterControl)
+		}
+	}
```

Comment:
I agree. There was little problem that I came across and how I'm going to still fix this inefficiency in the next commit.

I'm anticipating that in the future we will want to do more detailed gap analysis (e.g. show gap analysis for which controls are marked as "partially" or "not" implemented. [example](https://github.com/18F/cg-compliance/blob/a8c118a91ce61017c913948d4ebc063a0a556158/AC_Policy/component.yaml#L13)) In order to do that, I'm going to need the full object and not just unique strings. Which is all good but the type of object in `masterControlList` is of type `models.Control` and the ones given to us by the ones actually documented, `actualSatisfiedControls`, are of `models.Satisfies`. Which won't work out the box with set algebra.

New solution
```go
type Inventory struct {
	*models.OpenControl
	masterControlList       map[string]models.Control
	actualSatisfiedControls map[string]models.Satisfies
	MissingControlList      map[string]models.Control
}
```
The keys for `masterControlList` and `actualSatisfiedControls` are those unique strings with the `standard@control` but each map will contains their respective object as the value. I can do quick look ups and comparisons without the double for-loop and in the future we can pull out more details from each of the objects.